superset/views/annotations.py

```diff
@@ -18,12 +18,30 @@
 from flask_appbuilder.models.sqla.interface import SQLAInterface
 from flask_babel import gettext as __
 from flask_babel import lazy_gettext as _
+from wtforms.validators import StopValidation
 
 from superset import appbuilder
 from superset.models.annotations import Annotation, AnnotationLayer
 from .base import DeleteMixin, SupersetModelView
 
 
+class StartEndDttmValidator(object):
+    """
+    Validates dttm fields.
+    """
+    def __call__(self, form, field):
+        if not form['start_dttm'].data and not form['end_dttm'].data:
+            raise StopValidation(
+                _('annotation start time or end time is required.'),
```

Comment:
NIT: capitalize the `a`

Reply:
The final user message is: "Not a valid datetime value annotation start time or end time is required."